GetOps' getopt.c declarations clashes with system getopt declarations

This issue only occured (was discovered?) due to fluke with https://gitlab.sac-home.org/sac-group/sac2c/-/merge_requests/188. The following error message was issued:

```/root/stdlib/src/system/src/GetOpt/getopt.c:28:12: error: static declaration of 'optind' follows non-static declaration
   28 | static int optind = 1;
      |            ^~~~~~
In file included from /usr/include/x86_64-linux-gnu/bits/getopt_posix.h:27,
                 from /usr/include/unistd.h:869,
                 from /usr/include/x86_64-linux-gnu/hwloc/autogen/config.h:213,
                 from /usr/include/hwloc.h:56,
                 from /root/sac2c/build-test/include/libsac/hwloc/cpubind.h:9,
                 from /root/sac2c/build-test/include/runtime/hwloc_h/cpubind.h:42,
                 from /root/sac2c/build-test/include/sac.h:66,
                 from /root/stdlib/src/system/src/GetOpt/getopt.c:18:
/usr/include/x86_64-linux-gnu/bits/getopt_core.h:50:12: note: previous declaration of 'optind' was here
   50 | extern int optind;
      |            ^~~~~~
```
and more of the same for other static variables.

Possible solutions:
- Rename statics to SAC_getopt_optind or similar...
- Instead use system getopt **but** there might be a clash with how SaC related args are parsed otherwise.

--- a/src/system/src/GetOpt/getopt.c
+++ b/src/system/src/GetOpt/getopt.c
@@ -1,218 +1,187 @@
-/*
- * Implement the C library function getopt(3) for SAC.
- * Use classic short getopt without some of the GNU extensions.
- * I.e., it don't support the dash '-' and plus '+' as part of the optstring.
- * When an argument is part of an option, like for "-oarg", then
- * recognize "arg" is as an argument, not as an option.
- * See the getopt(3) manual page for details.
+/******************************************************************************
  *
- * Use 'optind' to specify the next argv to consider.
- * Use 'opterr' to enable/disable error reporting.
- * Use 'optopt' to return the most recent option.
- * Use 'optarg' to return the option argument, if any.
- * Use 'argind' to specify the next option as argv[optind][argind].
- */
-
+ * @brief Implements the C library function getopt(3) for SAC. Use classic short
+ * getopt without some of the GNU extensions. I.e., it don't support the dash
+ * '-' and plus '+' as part of the optstring. When an argument is part of an
+ * option, like for "-oarg", then recognize "arg" is as an argument, not as an
+ * option. See the getopt(3) manual page for details.
+ *
+ * @param SAC_optind: specify the next argv to consider.
+ * @param SAC_opterr: enable/disable error reporting.
+ * @param SAC_optopt: return the most recent option.
+ * @param SAC_optarg: return the option argument, if any.
+ * @param SAC_argind: specify the next option as argv[SAC_optind][SAC_argind].
+ *
+ ******************************************************************************/
 #include <string.h>
 #include <stdio.h>
+
 #include "sac.h"
 
-/*
- * Import the following two prototypes from SAC's commandline parser:
- *
- *     stdlib/world/system/src/CommandLine/CommandLine.c
- */
+/* Import the following two prototypes from SAC's commandline parser:
+ *  stdlib/world/system/src/CommandLine/CommandLine.c */
 int SACargc( void);
 char *SACargv(int n);
 
-static int optind = 1;
-static int opterr = 1;
-static int optopt;
-static char *optarg;
-
-static int argind = 1;
+static int SAC_optind = 1;
+static int SAC_opterr = 1;
+static int SAC_optopt;
+static char *SAC_optarg;
+static int SAC_argind = 1;
 
 int optEND(void)
 {
-  return '\0';
+    return '\0';
 }
 
 int get_optind(void)
 {
-  return optind;
+    return SAC_optind;
 }
 
 void set_optind(int set)
 {
-  optind = set;
-  argind = 1;
+    SAC_optind = set;
+    SAC_argind = 1;
 }
 
 int get_opterr(void)
 {
-  return opterr;
+    return SAC_opterr;
 }
 
 void set_opterr(int set)
 {
-  opterr = set;
+    SAC_opterr = set;
 }
 
 int get_optopt(void)
 {
-  return optopt;
+    return SAC_optopt;
 }
 
 static inline char *my_strdup(const char *str)
 {
-  char *copy = SAC_MALLOC(strlen(str) + 1);
-
-  strcpy(copy, str);
-  return copy;
+    char *copy = SAC_MALLOC(strlen(str) + 1);
+    strcpy(copy, str);
+    return copy;
 }
 
 char *get_optarg(void)
 {
-  return optarg ? my_strdup(optarg) : my_strdup("");
+    return SAC_optarg ? my_strdup(SAC_optarg) : my_strdup("");
 }
 
 static void badopt(void)
 {
-  if (opterr)
-  {
-    char *prog = SACargv(0);
-
-    fprintf(stderr, "%s: invalid option -- '%c'\n", prog, optopt);
-    free(prog);
-  }
+    if (SAC_opterr) {
+        char *prog = SACargv(0);
+        fprintf(stderr, "%s: invalid option -- '%c'\n", prog, SAC_optopt);
+        free(prog);
+    }
 }
 
 static void setarg(char *arg)
 {
-  if (optarg)
-  {
-    free(optarg);
-  }
-  optarg = arg;
+    if (SAC_optarg != NULL) {
+        free(SAC_optarg);
+    }
+
+    SAC_optarg = arg;
 }
 
 static void missing( void)
 {
-  if (opterr == 1)
-  {
-    char *prog = SACargv(0);
-
-    fprintf(stderr, "%s: option requires an argument -- '%c'\n", prog, optopt);
-    free(prog);
-  }
+    if (SAC_opterr == 1) {
+        char *prog = SACargv(0);
+        fprintf(stderr, "%s: option requires an argument -- '%c'\n", prog, SAC_optopt);
+        free(prog);
+    }
 }
 
 int getopt_sac(const char *opts)
 {
-  const int argc = SACargc();
-  int retval;
+    const int argc = SACargc();
+    int retval;
 
-  setarg(NULL);
+    setarg(NULL);
 
-  if (optind < 1)
-  {
-    optind = 1;
-    argind = 1;
-  }
-
-  if (optind == 1 && argind == 1 && optind < argc)
-  {
-    char *arg = SACargv(optind);
-    /* SAC uses the "-mt" option to enable multi-threading. */
-    if (!strcmp(arg, "-mt"))
-    {
-      ++optind;
+    if (SAC_optind < 1) {
+        SAC_optind = 1;
+        SAC_argind = 1;
     }
-    free(arg);
-  }
 
-  if (optind >= argc)
-  {
-    retval = optEND();
-  }
-  else
-  {
-    char *arg = SACargv(optind);
+    if (SAC_optind == 1 && SAC_argind == 1 && SAC_optind < argc) {
+        char *arg = SACargv(SAC_optind);
 
-    if (arg[0] != '-')
-    {
-      retval = optEND();
-    }
-    else if (arg[1] == '\0')
-    {
-      retval = optEND();
-    }
-    else if (arg[1] == '-' && arg[2] == '\0')
-    {
-      optind++;
-      argind = 1;
-      retval = optEND();
-    }
-    else
-    {
-      if ((size_t)argind >= strlen(arg))
-      {
-        optopt = '\0';
-      }
-      else
-      {
-        optopt = arg[argind];
-      }
-      char *p;
-
-      if (optopt == ':' ||
-          optopt == '\0' ||
-          (p = strchr(opts, optopt)) == NULL)
-      {
-        badopt();
-        if (arg[++argind] == '\0')
-        {
-          ++optind;
-          argind = 1;
+        // SaC uses the "-mt" option to enable multi-threading.
+        if (!strcmp(arg, "-mt")) {
+            SAC_optind += 1;
         }
-        retval = '?';
-      }
-      else if (*++p == ':')
-      {
-        if (arg[++argind] != '\0')
-        {
-          setarg(my_strdup(&arg[argind]));
-          ++optind;
-          argind = 1;
-          retval = optopt;
-        }
-        else if (++optind >= argc)
-        {
-          missing();
-          argind = 1;
-          retval = '?';
-        }
-        else
-        {
-          setarg(SACargv(optind));
-          ++optind;
-          argind = 1;
-          retval = optopt;
-        }
-      }
-      else if (arg[++argind] == '\0')
-      {
-        ++optind;
-        argind = 1;
-        retval = optopt;
-      }
-      else
-      {
-        retval = optopt;
-      }
-    }
-    free(arg);
-  }
 
-  return retval;
+        free(arg);
+    }
+
+    if (SAC_optind >= argc) {
+        retval = optEND();
+    } else {
+        char *arg = SACargv(SAC_optind);
+
+        if (arg[0] != '-') {
+            retval = optEND();
+        } else if (arg[1] == '\0') {
+            retval = optEND();
+        } else if (arg[1] == '-' && arg[2] == '\0') {
+            SAC_optind += 1;
+            SAC_argind = 1;
+            retval = optEND();
+        } else {
+            if ((size_t)SAC_argind >= strlen(arg)) {
+                SAC_optopt = '\0';
+            } else {
+                SAC_optopt = arg[SAC_argind];
+            }
+
+            char *p;
+
+            if (SAC_optopt == ':' ||
+                SAC_optopt == '\0' ||
+                (p = strchr(opts, SAC_optopt)) == NULL) {
+
+                badopt();
+                if (arg[++SAC_argind] == '\0') {
+                    SAC_optind += 1;
+                    SAC_argind = 1;
+                }
+
+                retval = '?';
+            } else if (*++p == ':') {
+                if (arg[++SAC_argind] != '\0') {
+                    setarg(my_strdup(&arg[SAC_argind]));
+                    SAC_optind += 1;
+                    SAC_argind = 1;
+                    retval = SAC_optopt;
+                } else if (++SAC_optind >= argc) {
+                    missing();
+                    SAC_argind = 1;
+                    retval = '?';
+                } else {
+                    setarg(SACargv(SAC_optind));
+                    ++SAC_optind;
+                    SAC_argind = 1;
+                    retval = SAC_optopt;
+                }
+            } else if (arg[++SAC_argind] == '\0') {
+                SAC_optind += 1;
+                SAC_argind = 1;
+                retval = SAC_optopt;
+            } else {
+                retval = SAC_optopt;
+            }
+        }
+
+        free(arg);
+    }
+
+    return retval;
 }

